classes/cmake.yaml

```diff
@@ -71,27 +76,73 @@ buildScript: |
                     INSTALL_OPTIONS+=( "$OPTARG" )
                     ;;
                 \?)
-                  echo "Invalid option: -$OPTARG" >&2
-                  exit 1
-                  ;;
+                    echo "Invalid option: -$OPTARG" >&2
+                    exit 1
+                    ;;
             esac
         done
         shift $(( OPTIND -1 ))
 
         mkdir -p build install
         pushd build
+
+        # windows does not support DESTDIR, we have to use full-path
+        # for CMAKE_INSTALL_PREFIX instead
+        DESTDIR="${PWD}/../install"
+        INSTALL_PREFIX="/usr"
+
+        OPTIONS=( ${CMAKE_GENERATOR:+-G"${CMAKE_GENERATOR}"} )
+        if [[ ${CMAKE_GENERATOR} == "Visual Studio"* ]] ; then
+            INSTALL_PREFIX="$DESTDIR$INSTALL_PREFIX"
+            # since VS2019 we can select the platform and target by special variables
+            OPTIONS+=( ${CMAKE_GENERATOR_PLATFORM:+-A"${CMAKE_GENERATOR_PLATFORM}"} )
+            OPTIONS+=( ${CMAKE_GENERATOR_TOOLSET:+-T"${CMAKE_GENERATOR_TOOLSET}"} )
+        else
+            # windows does not support a sigle build type, we have to select the
+            # current type with cmake --build . --config
+            OPTIONS+=( -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-RelWithDebInfo} )
```

Comment:
Here I'll investigate again, I guess, I also could set the BUILD_TYPE to bob for Windows, if I added this configuration to `CMAKE_CONFIGURATION_TYPES`.